docs/source/quilt_proposal.md

```diff
@@ -0,0 +1,324 @@
+
+Single frame mutations (`SET-FREQUENCY`, `SHIFT-FREQUENCY`, `SET-PHASE`, `SHIFT-PHASE`,
+`SET-SCALE`) have a hardware dependent duration (which may be effectively zero).
+These operations block pulses on the targeted frame.
+
+The `SWAP-PHASE` instruction introduces an implicit synchronization on the two
+involved frames. In other words, any operations involving either of the swapped
+frames and preceding the `SWAP-PHASE` must complete prior to the `SWAP-PHASE`
+event.
+
+### Calibrations
+
+Calibrations can be associated with gates in Quil to aid the compiler in
+converting a list of gates into the corresponding series of pulses.
+
+Calibrations can be parameterized and include concrete values, which are
+resolved in "Haskell-style", with later definitions being prioritized over
+earlier ones. For example, given the following list of calibration definitions
+in this order:
+1. `DEFCAL RX(%theta) %qubit:`
+2. `DEFCAL RX(%theta) 0:`
+3. `DEFCAL RX(pi/2) 0:`
+The instruction `RX(pi/2) 0` would match (3), the instruction `RX(pi) 0` would
+match (2), and the instruction `RX(pi/2) 1` would match (1).
+
+The body of a DEFCAL is a list of analog control instructions that ideally
+enacts the corresponding gate.
+
+
+## Hypothetical Examples
+
+Here are some example calibrations for various types of gates and measurements.
+
+Setting up frequencies:
+```
+SET-FREQUENCY 0 "xy" 4678266018.71412
+SET-FREQUENCY 1 "xy" 3821271416.79618
+
+SET-FREQUENCY 0 1 "cz" 137293415.829024
+
+SET-FREQUENCY 0 "ro" 5901586914.0625
+SET-FREQUENCY 1 "ro" 5721752929.6875
+
+SET-FREQUENCY 0 "out" 5901586914.0625
+SET-FREQUENCY 1 "out" 5721752929.6875
+```
+
+Calibrations of RX:
+```
+DEFCAL RX(%theta) 0:
+    SET-SCALE %theta/pi*0.936
+    PULSE 0 "xy" draggaussian(duration: 80e-9, fwhm: 40e-9, t0: 40e-9, anh: -210e6, alpha: 0)
+
+DEFCAL RX(pi/2) 0:
+    SET-SCALE 0.468
+    PULSE 0 "xy" draggaussian(duration: 80e-9, fwhm: 40e-9, t0: 40e-9, anh: -210e6, alpha: 0)
+
+# With crosstalk mitigation - no pulses on neighbors
+DEFCAL RX(pi/2) 0:
+    FENCE 0 1 7
+    PULSE 0 "xy" draggaussian(duration: 80e-9, fwhm: 40e-9, t0: 40e-9, anh: -210e6, alpha: 0)
+    FENCE 0 1 7
+```
+
+RZ:
+```
+DEFCAL RZ(%theta) %qubit:
+    # RZ of +theta corresponds to a frame change of -theta
+    SHIFT-PHASE %qubit "xy" -%theta
+```
+
+Calibrations of CZ:
+```
+DEFCAL CZ 0 1:
+    PULSE 0 1 "cz" erfsquare(duration: 340e-9, risetime: 20e-9, padleft: 8e-9, padright: 8e-9)
+    SHIFT-PHASE 0 "xy" 0.00181362669
+    SHIFT-PHASE 1 "xy" 3.44695296672
+
+# With no parallel 2Q gates
+DEFCAL CZ 0 1:
+    FENCE
+    PULSE 0 1 "cz" erfsquare(duration: 340e-9, risetime: 20e-9, padleft: 8e-9, padright: 8e-9)
+    SHIFT-PHASE 0 "xy" 0.00181362669
+    SHIFT-PHASE 1 "xy" 3.44695296672
+    FENCE
+```
+
+Readout:
+```
+DEFCAL MEASURE 0 dest:
+    DECLARE iq REAL[2]
+    PULSE 0 "ro" flat(duration: 1.2e-6, iq: ???)
+    CAPTURE 0 "out" flat(duration: 1.2e-6, iq: ???) iq
+    LT %dest iq[0] ??? # thresholding
```

Comment:
- [ ] LT in the grammar but not described in the spec (that I see)
- [ ] ??? should be filled in

Reply:
The typed memory proposal is supported by qvm, but not necessarily hardware, and not the translator.

It should be emphasized that these are illustrative examples and the values should not be taken to be meaningful.

The `LT` should be replaced with a `PRAGMA FILTER-NODE` to use language supported in translation.